packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -112,15 +114,17 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         !areEqual(prevIDs, nextIDs)
       ) {
         this._release();
-        this._localVariables = null;
+        this._localVariables = this._getFragmentVariables();
         this._resolver = createFragmentSpecResolver(
           relay,
           fragments,
           nextProps,
           this._handleFragmentDataUpdate,
         );
         this.setState({relayProp: this._buildRelayProp(relay)});
-      } else if (!this._localVariables) {
+      } else if (true && !this._localVariables) {
+        // question: why did we only call `setProps` if contains hasn't refetched?
+        //           shell we always keep _resolver updated no matter whehter comtains has refetched or not?        
```

Comment:
Good question. `localVariables` is null when the container is using variables assigned from above, and non-null once data has been refetched. The prev/next ID check above ensures that if new props refer to different objects completely that we *will* update props. If the new props refer to the same record - this case - then by definition we have already updated the resolver to the correct ids & variables (when the refetch occurred).